### DISCUSSION TITLE
C vs C++ interface

### DISCUSSION THREAD

**Discussion by rayegun** [Z3Prover/z3#5619] [Q&A] [answered]
Hi, just a quick question.

Other than usability improvements from using C++ are there any capabilities found in one but not the other? I'd like to use C if it is a complete interface.

Thanks

**Comment by NikolajBjorner** [comment, marked as answer]
C is _the_ complete interface. The C++ interface builds on top of the C interface. All other interfaces also build on top of the C interface. It is the only binary interface to Z3. You can also call the C interface from C++ code that uses the C++ interface because it is already included.
